src/chainparams.cpp

```diff
@@ -179,9 +181,9 @@ class CTestNetParams : public CMainParams {
 
         checkpointData = (Checkpoints::CCheckpointData) {
             boost::assign::map_list_of
-            ( 546, uint256S("000000002a936ca763904c3c35fce2f3556c559c0214345d31b1bcebf76acb70")),
```

Comment:
In fact, why not add the genesis checkpoint (the genesis block, the only checkpoint that cannot be reorged) to the mainchain too (and while at it use consensus.hashGenesisBlock for  regtest's gensis checkpoint for uniformity).